lib/backend/dynamo/shards.go

```diff
@@ -83,26 +83,70 @@ func (b *Backend) pollStreams(externalCtx context.Context) error {
 	set := make(map[string]struct{})
 	eventsC := make(chan shardEvent)
 
-	refreshShards := func() error {
+	shouldStartPoll := func(shard *dynamodbstreams.Shard) bool {
+		sid := aws.StringValue(shard.ShardId)
+		if _, ok := set[sid]; ok {
+			// already being polled
+			return false
+		}
+		if _, ok := set[aws.StringValue(shard.ParentShardId)]; ok {
+			b.Warnf("Skipping child shard: %s, still polling parent %s", sid, aws.StringValue(shard.ParentShardId))
+			// still processing parent
+			return false
+		}
+		return true
+	}
+
+	refreshShards := func(init bool) (bool, error) {
```

Comment:
Nit: Could you explain either with a comment or named return args what the bool return argument indicates?

Reply:
Bool flag has been removed.  It wasn't really necessary anymore.